src/zenml/artifacts/utils.py

```diff
@@ -183,7 +188,9 @@ def save_artifact(
     response = Client().zen_store.create_artifact(artifact=artifact)
     if artifact_metadata:
         Client().create_run_metadata(
-            metadata=artifact_metadata, artifact_id=response.id
+            metadata=artifact_metadata,
+            resource_id=response.id,
+            resource_type=MetadataResourceTypes.ARTIFACT,
```

Comment:
The changes in this file will also be affected by [the PR here](https://github.com/zenml-io/zenml/pull/2081). We have to coordinate the efforts between you and @fa9r.